sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -598,6 +599,22 @@ int gnrc_ndp_internal_sl2a_opt_handle(gnrc_pktsnip_t *pkt, ipv6_hdr_t *ipv6, uin
         pkt = pkt->next;
     }
 
+#ifdef MODULE_GNRC_SIXLOWPAN_ND
+    if ((sl2a_len == 2) || (sl2a_len == 8)) {
+        // The link-layer seems to be IEEE 802.15.4.
+        // Determining address length from the option length:
+        // https://tools.ietf.org/html/rfc4944#section-8
+        if (sl2a_opt->len == 1) {
+            sl2a_len = 2;
+        } else if (sl2a_opt->len == 2) {
+            sl2a_len = 8;
+        } else {
```

Comment:
This needs some style fixes (`/* .. */` instead of `//` and newline before `else`).